components/media-upload-button/index.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * WordPress dependencies
+ */
+import { Component } from 'element';
+import { __ } from 'i18n';
+import Button from 'components/button';
+
+class MediaUploadButton extends Component {
```

Comment:
I'd put this at the same abstraction level as `Editable`, not in general UI components.